src/logic/safe/store/reducer/__tests__/gatewayTransactions.test.ts

```diff
@@ -399,386 +247,11 @@ describe('gatewayTransactionsReducer', () => {
         4: {
           [ZERO_ADDRESS]: {
             queued: {
-              next: {},
-              queued: {
+              next: {
                 0: [mockTx.transaction],
                 1: [mockTx2.transaction],
                 2: [mockTx3.transaction],
               },
-            },
-            history: {},
-          },
-        },
-      }
-
-      expect(newState).toEqual(expectedState)
-    })
-
-    it('merges existing `queued.queued` transactions', () => {
-      const mockTx: Transaction = {
-        transaction: { id: 'gsdfgsdfgfdgsf', ...MOCK_TX_SUMMARY },
-        ...MOCK_TX_META,
-      }
-
-      const mockPayload: QueuedPayload = {
-        chainId: '4',
-        safeAddress: ZERO_ADDRESS,
-        values: [QUEUED_LABEL, mockTx],
-      }
-
-      const prevState: GatewayTransactionsState = {
-        4: {
-          [ZERO_ADDRESS]: {
-            queued: {
-              next: {},
-              queued: {
-                0: [
-                  {
-                    ...mockTx.transaction,
-                    // Would have been added via `UPDATE_TRANSACTION_DETAILS`
-                    txDetails: {
-                      txId: mockTx.transaction.id,
-                      txStatus: mockTx.transaction.txStatus,
-                      ...MOCK_TX_DETAILS,
-                    },
-                  },
-                ],
-              },
-            },
-            history: {},
-          },
-        },
-      }
-
-      const newState = gatewayTransactionsReducer(prevState, addQueuedTransactions(mockPayload))
-
-      // Should remain the same as it keeps the `txDetails` via merge
-      expect(newState).toEqual(prevState)
-    })
-
-    it('sorts `queued.queued` transactions by ascending timestamp when adding a new transaction', () => {
```

Comment:
This needs to happen as the batched transaction execution takes the first transaction in the array. It is also more consistent in the UI to organise them.

Maybe we should add fallback logic in the batching though as well. cc @usame-algan

Reply:
Good idea! We can keep the transactions unsorted in the store and sort them on demand via selectors.

Reply:
I would still sort them in the store.

Reply:
Not always but sometimes.

Reply:
I don't think it's a good idea to mess with the backend sort. If they aren'r chronological and you think they should be, it needs to be fixed in the API.

Reply:
@usame-algan, make sure you add the ordering to the batch execution then.